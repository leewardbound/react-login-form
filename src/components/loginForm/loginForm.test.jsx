import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import LoginForm from '.';

const mockStore = configureMockStore([ thunk ]);

describe('LoginForm', function () {
  it('should render correctly with no error', function () {
    const store = mockStore({ auth: { error: false } });

    const component = renderer.create(
      <LoginForm store={store} />
    );

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with auth error', function () {
    const store = mockStore({ auth: { error: true } });

    const component = renderer.create(
      <LoginForm store={store} />
    );

    expect(component).toMatchSnapshot();
  });
});
