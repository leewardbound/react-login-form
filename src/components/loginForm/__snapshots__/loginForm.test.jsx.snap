// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render correctly with auth error 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <p
    className="fJygQn"
    type="error"
  >
    Invalid email or password
  </p>
  <div
    className="bAxRVC"
  >
    <div
      className="gxCkEM"
    >
      <label
        className="gGZfep"
        htmlFor="email"
      >
        Email:
      </label>
      <input
        className="jpmZcP"
        id="email"
        name="email"
        onChange={[Function]}
        type="email"
      />
    </div>
    <div
      className="gxCkEM"
    >
      <label
        className="gGZfep"
        htmlFor="password"
      >
        Password:
      </label>
      <input
        className="jpmZcP"
        id="password"
        name="password"
        onChange={[Function]}
        type="password"
      />
    </div>
  </div>
  <div
    className="bAxRVC"
  >
    <div
      className="hmkMuH"
    >
      <label
        className="ieGGlk"
        htmlFor="remember"
      >
        <input
          id="remember"
          name="remember"
          type="checkbox"
        />
        Remember me
      </label>
      <button
        className="dVTSea"
        type="submit"
      >
        Log in
      </button>
    </div>
  </div>
</form>
`;

exports[`LoginForm should render correctly with no error 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="bAxRVC"
  >
    <div
      className="gxCkEM"
    >
      <label
        className="gGZfep"
        htmlFor="email"
      >
        Email:
      </label>
      <input
        className="jpmZcP"
        id="email"
        name="email"
        onChange={[Function]}
        type="email"
      />
    </div>
    <div
      className="gxCkEM"
    >
      <label
        className="gGZfep"
        htmlFor="password"
      >
        Password:
      </label>
      <input
        className="jpmZcP"
        id="password"
        name="password"
        onChange={[Function]}
        type="password"
      />
    </div>
  </div>
  <div
    className="bAxRVC"
  >
    <div
      className="hmkMuH"
    >
      <label
        className="ieGGlk"
        htmlFor="remember"
      >
        <input
          id="remember"
          name="remember"
          type="checkbox"
        />
        Remember me
      </label>
      <button
        className="dVTSea"
        type="submit"
      >
        Log in
      </button>
    </div>
  </div>
</form>
`;
